cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -567,19 +598,61 @@ open class Node @Inject constructor(private val project: Project) {
+
+        // TODO Refactor this code to extract the information from the config file
+        val p2pHost = config.getString("p2pAddress")
+        this.p2pPort = p2pHost.substring(p2pHost.indexOf(":") + 1).toInt()
+        val rpcConfig = config.getString("rpcSettings.address")
+        this.rpcPort = rpcConfig.substring(rpcConfig.indexOf(":") + 1).toInt()
+        val rpcAdminConfig = config.getString("rpcSettings.adminAddress")
+        val rpcAdminPort = rpcAdminConfig.substring(rpcConfig.indexOf(":") + 1)
+
         val dockerConf = config
-            .withValue("p2pAddress", ConfigValueFactory.fromAnyRef("$containerName:$p2pPort"))
-            .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.port}"))
-            .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.adminPort}"))
-            .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
-            .withFallback(configDefaults)
+                .withValue("dataSourceProperties.dataSource.url", ConfigValueFactory.fromAnyRef("jdbc:h2:file:./persistence/persistence;DB_CLOSE_ON_EXIT=FALSE;WRITE_DELAY=0;LOCK_TIMEOUT=10000"))
+                .withValue("p2pAddress", ConfigValueFactory.fromAnyRef("$containerName:$p2pPort"))
+                .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:$rpcPort"))
+                .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:$rpcAdminPort"))
+                .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
 
         config = dockerConf
         createNodeAndWebServerConfigFiles(config)
     }
 
+     /**
+     * Installs the Dockerized configuration file to the root directory and detokenises it.
+     *
+     * Defaults to Postgres (part of EG-496 Epic)
+     */
+    internal fun installDatabaseConfig(port: Int, address: String, initConfig: String) {
+        installResource(initConfig)
+        setDBSettings(port, address)
+
+        val dockerConf = config
+                .withValue("dataSourceProperties.dataSourceClassName", ConfigValueFactory.fromAnyRef("org.postgresql.ds.PGSimpleDataSource"))
+                .withValue("dataSourceProperties.dataSource.url", ConfigValueFactory.fromAnyRef("jdbc:postgresql://$address:$port/mydb?currentSchema=myschema"))
+                .withValue("dataSourceProperties.dataSource.user", ConfigValueFactory.fromAnyRef("myuser"))
+                .withValue("dataSourceProperties.dataSource.password", ConfigValueFactory.fromAnyRef("mypassword"))
+                .withValue("database.transactionIsolationLevel", ConfigValueFactory.fromAnyRef("READ_COMMITTED"))
+                .withValue("database.runMigration", ConfigValueFactory.fromAnyRef(true))
+                .withValue("database.schema", ConfigValueFactory.fromAnyRef("myschema"))
```

Comment:
Again there's no fallback to defaults here, which means anything you don't specify will be unset.

Reply:
There is no need for the fallback because all those properties are specified. Even if they are specified they will be overwritten